src/engine/guidance/post_processing.cpp

```diff
@@ -1294,33 +1299,36 @@ std::vector<RouteStep> collapseUseLane(std::vector<RouteStep> steps)
         return index;
     };
 
-    const auto canCollapeUseLane =
-        [containsTag](const util::guidance::LaneTupel lanes,
-                      extractor::guidance::TurnLaneDescription lane_description) {
-            // the lane description is given left to right, lanes are counted from the right.
-            // Therefore we access the lane description using the reverse iterator
-            if (lanes.first_lane_from_the_right > 0 &&
-                containsTag(*(lane_description.rbegin() + (lanes.first_lane_from_the_right - 1)),
-                            (extractor::guidance::TurnLaneType::straight |
-                             extractor::guidance::TurnLaneType::none)))
-                return false;
-
-            const auto lane_to_the_right = lanes.first_lane_from_the_right + lanes.lanes_in_turn;
-            if (lane_to_the_right < boost::numeric_cast<int>(lane_description.size()) &&
-                containsTag(*(lane_description.rbegin() + lane_to_the_right),
-                            (extractor::guidance::TurnLaneType::straight |
-                             extractor::guidance::TurnLaneType::none)))
-                return false;
+    const auto canCollapseUseLane = [containsTag](
+        const util::guidance::LaneTuple lanes,
+        const extractor::guidance::TurnLaneDescription lane_description) {
+        // the lane description is given left to right, lanes are counted from the right.
+        // Therefore we access the lane description using the reverse iterator
+        using iter_type = extractor::guidance::TurnLaneDescription::const_iterator;
+
+        boost::iterator_range<iter_type> right_most_lanes =
+            extractor::guidance::lanesToTheRight<iter_type>(lanes, lane_description);
```

Comment:
This is the ugly callsite we want to spare the user from having to write. The compiler should be able to deduce the type if you work out the notes above.